source/base/logstream.cc

```diff
@@ -389,18 +390,9 @@ LogStream::get_prefixes() const
   // from the initial thread that created logstream.
   if (!exists)
     {
-      const tbb::enumerable_thread_specific<std::stack<std::string>> &impl =
-        prefixes.get_implementation();
-
-      // The thread that created this LogStream object should be the first
-      // in tbb's enumerable_thread_specific container.
-      const tbb::enumerable_thread_specific<
-        std::stack<std::string>>::const_iterator first_elem = impl.begin();
-
-      if (first_elem != impl.end())
-        {
-          local_prefixes = *first_elem;
-        }
+      auto it = prefixes.data.find(master_thread);
+      if (it != prefixes.data.end())
+        local_prefixes = it->second;
```

Comment:
@bangerth Actually, the fix isn't too horrible. It just requires an ugly `friend` declaration.

Reply:
It's not thread-safe to do this, but neither was the original implementation...

Reply:
Yeah. But I won't fix that. :smiling_imp: